File: src/test/java/com/morak/performancetracker/aop/persistence/QueryMonitorAopTest.java
package com.morak.performancetracker.aop.persistence;

import static org.assertj.core.api.Assertions.assertThat;

import java.lang.reflect.Proxy;
import java.sql.Connection;
import java.sql.SQLException;
import javax.sql.DataSource;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
class QueryMonitorAopTest {

    private final DataSource dataSource;

    @Autowired
    public QueryMonitorAopTest(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    @Test
    void 커넥션을_프록시로_만든다() throws SQLException {
        // given
        Connection proxyConnection = dataSource.getConnection();
        // when
        Class<? extends Connection> proxyClass = proxyConnection.getClass();
        // then
        assertThat(Proxy.isProxyClass(proxyClass)).isTrue();
    }
}

File: src/test/kotlin/com/morak/performancetracker/aop/persistence/QueryMonitorAopTest.kt
package com.morak.performancetracker.aop.persistence

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import java.lang.reflect.Proxy
import java.sql.Connection
import java.sql.SQLException
import javax.sql.DataSource

@SpringBootTest
internal class QueryMonitorAopTest @Autowired constructor(private val dataSource: DataSource) {
    @Test
    @Throws(SQLException::class)
    fun 커넥션을_프록시로_만든다() {
        // given
        val proxyConnection = dataSource.connection
        // when
        val proxyClass: Class<out Connection> = proxyConnection.javaClass
        // then
        Assertions.assertThat(Proxy.isProxyClass(proxyClass)).isTrue()
    }
}
